src/Plug.vala

```diff
@@ -36,22 +56,120 @@ namespace Power {
         }
 
         public override Gtk.Widget get_widget () {
-            if (main_view == null) {
-                main_view = new MainView ();
+            if (main_grid == null) {
+                manufacturer_logo = new Gtk.Image () {
+                    halign = Gtk.Align.END,
+                    pixel_size = 32,
+                    use_fallback = true
+                };
+
+                var fileicon = new FileIcon (File.new_for_path (manufacturer_icon_path));
+                if (manufacturer_icon_path != null) {
+                    manufacturer_logo.gicon = fileicon;
+                }
+
+                if (manufacturer_logo.gicon == null) {
+                    load_fallback_manufacturer_icon.begin ();
+                }
+
+                stack = new Gtk.Stack ();
+                main_view = new BehaviorView ();
+                stack.add_named (main_view, "Power");
+
+                if (main_view.battery.is_present ()) {
+                    var badge_icon = new Gtk.Image.from_icon_name (main_view.battery.get_icon_name_for_battery (), Gtk.IconSize.BUTTON) {
+                        halign = Gtk.Align.END,
+                        valign = Gtk.Align.END
+                    };
+
+                    var overlay = new Gtk.Overlay ();
+                    overlay.add (manufacturer_logo);
+                    overlay.add_overlay (badge_icon);
+                    device_view = new DeviceView (
+                        main_view.battery,
+                        overlay,
+                        main_view.battery.get_info (),
+                        "Built-in",
```

Comment:
I think this makes it more clear we're specifically talking about a battery. Plus, make translatable.

```suggestion
                        _("Built-in Battery"),
```